move/axelar_gateway/sources/types/proof.move

```diff
@@ -64,45 +65,46 @@ public(package) fun recover_pub_key(
 /// Validates the signatures of a message against the signers.
 /// The total weight of the signatures must be greater than or equal to the threshold.
 /// Otherwise, the error `ELowSignaturesWeight` is raised.
-public(package) fun validate(
-    self: &Proof,
-    message: vector<u8>,
-) {
-    let signers = self.signers();
-    let signatures = self.signatures();
+public(package) fun validate(self: &Proof, message: vector<u8>) {
+    let signers = &self.signers;
+    let signatures = &self.signatures;
     assert!(signatures.length() != 0, ELowSignaturesWeight);
 
     let threshold = signers.threshold();
+    let signatures_length = signatures.length();
     let mut total_weight: u128 = 0;
     let mut signer_index = 0;
     let mut i = 0;
 
-    while(i < signatures.length()) {
-        let signature = signatures[i];
-
-        let pub_key = signature.recover_pub_key(&message);
+    while (i < signatures_length) {
+        let pub_key = signatures[i].recover_pub_key(&message);
 
-        let (weight, index) = find_weight_by_pub_key_from(signers, signer_index, &pub_key);
+        let (weight, index) = find_weight_by_pub_key_from(
+            signers,
+            signer_index,
+            &pub_key,
+        );
 
-        signer_index = index;
+        assert!(signer_index < signatures_length, EMalformedSigners);
```

Comment:
this is being done in the [find](https://github.com/axelarnetwork/axelar-cgp-sui/pull/140/files/e406dfb6fa0b8faa17d32d99cd45bc799ee88c49..37861f3761282d252ac73e3aecc7b7c6a498d2df#diff-92469ab1c1944669420bbcbf13715f8cbc924ce9ff1a0b2de4da86906a67a1e6R117) helper method